### PR TITLE
Add events support

### PR DESCRIPTION
**Pull request by jnschaeffer**
This PR adds support for `CreateRelationships` and `DeleteRelationships` in iam-runtime-infratographer using the Infratographer `events` package. This functionality makes iam-runtime-infratographer a fully featured implementation of the IAM runtime spec.

### CODE REVIEW

File: config.example.yaml [hunk 1]
```diff
@@ -5,5 +5,10 @@ permissions:
 jwt:
   jwksuri: https://identity-api.enterprise.dev/jwks.json
   issuer: https://identity-api.enterprise.dev/
+events:
+  nats:
+    url: localhost:4222
```
Comment (mikemrm) [inline config.example.yaml:10]: Does this need the protocol?
```suggestion
    url: nats://localhost:4222
```

File: internal/server/server.go [hunk 2]
```diff
@@ -147,3 +160,83 @@ func (s *server) CheckAccess(ctx context.Context, req *authorization.CheckAccess
...
+	s.logger.Infow("request", "req", authReq)
+
+	authResp, err := s.publisher.PublishAuthRelationshipRequest(ctx, authReq)
+	if err != nil {
+		return err
+	}
+
+	if authResp.Error() != nil {
+		return err
+	}
```
Comment (mikemrm) [inline internal/server/server.go:209]: This should be capturing the error, as the `err` value is already checked on line 203.
```suggestion
	if err := authResp.Error(); err != nil {
		return err
	}
```

**Review by mikemrm — CHANGES_REQUESTED** [review]
small fix, otherwise looks good

File: internal/server/server.go (hunk 2 above)
Comment (jnschaeffer) [reply internal/server/server.go:209]: Good catch!

File: config.example.yaml (hunk 1 above)
Comment (jnschaeffer) [reply config.example.yaml:10]: Probably good to add it regardless.

**Review by mikemrm — APPROVED** [review]

**Pull request merged**